conventions.md

```diff
@@ -0,0 +1,210 @@
+# Repository overview
+
+- **Project name:** LiFi  
+- **Purpose:** LiFi is a cross-chain bridge aggregation protocol that ensures secure and efficient interoperability through robust smart contracts and automated processes.  
+- **Core components:**
+  - **Smart contracts:** Primarily built using the Diamond Standard (EIP-2535) with modular facets for core functionality, along with supporting periphery contracts for extended features and integrations.
+  - **Automation scripts:** Deployment, migration, and operational tasks.
+  - **Tests:** Tests ensuring contract reliability and safety.
+  - **Documentation:** Detailed guides, API specifications, and deployment instructions.
+
+# Codebase structure
+
+    /lifi
+     ├── src/               # Solidity smart contracts (Diamond facets + periphery) - `.sol` files
+     ├── tasks/             # Utility scripts or tasks
+     ├── scripts/           # Deployment, migration, and automation scripts
+     ├── tests/             # Unit and integration tests - `.t.sol` files
+     ├── docs/              # Project documentation, API specs, and guides
+     ├── .github/           # Github workflows
+     ├── config/            # Facets' configuration files
+     ├── deployments/       # Deployment logs and addresses
+     ├── audit/             # Audit reports and log
+     ├── README.md          # High-level project overview and setup instructions
+     ├── conventions.md     # Repository conventions and guidelines (this file)
+
+Follow the folder structure to locate resources and generate or modify code in accordance with these standards.
+
+# Smart contract conventions
+
+## Solidity standards and patterns
+
+- **Solidity version:**  
+  All Solidity files must start with:
+
+    pragma solidity ^0.8.17;
+
+- **Design patterns:**  
+  - Use established patterns (e.g., Ownable for access control, Diamond Standard for facets).  
```

Comment:
"Diamond standard for facets"....not sure what that means tbh...

Reply:
I meant "Diamond standard with facets" but "EIP-2535 Diamond Standard for facet-based architectures" sounds more decent